src/views/AuthenticationWrapper/DevSecurity.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { ReactEventHandler, useEffect, useRef, useState } from 'react';
+import { OktaContext } from '@okta/okta-react';
+
+const storageKey = 'dev-user-config';
+
+const initialAuthState = {
+  isAuthenticated: false,
+  isPending: false,
+  name: '',
+  euaId: '',
+  groups: [] as string[]
+};
+
+const jobCodes = ['EASI_D_GOVTEAM'];
+
+type ParentComponentProps = {
+  children: React.ReactNode;
+};
+
+const DevSecurity = ({ children }: ParentComponentProps) => {
+  const [authState, setAuthState] = useState(initialAuthState);
+  const [euaId, setEuaId] = useState('');
+  const checkboxValues = useRef(new Set<string>(jobCodes));
+
+  const authService = {
+    login: () => {},
+    logout: () => {
+      window.localStorage.removeItem(storageKey);
+      window.location.href = '/';
+    },
+    getUser: () => {
+      return Promise.resolve({ name: authState.name });
+    },
+    getTokenManager: () => {
+      return {
+        off: () => {},
+        on: () => {},
+        renew: () => {}
+      };
+    }
+  };
+
+  useEffect(() => {
+    if (window.localStorage[storageKey]) {
+      const state = JSON.parse(window.localStorage[storageKey]);
+      setAuthState(as => {
+        return {
+          ...as,
+          name: `User ${state.eua}`,
+          isAuthenticated: true,
+          euaId: state.eua,
+          groups: state.jobCodes
+        };
+      });
+    }
+  }, []);
+
+  const handleSubmit: ReactEventHandler = event => {
+    event.preventDefault();
+    const value = {
+      eua: euaId,
+      jobCodes: Array.from(checkboxValues.current)
+    };
+    localStorage.setItem(storageKey, JSON.stringify(value));
+    localStorage.removeItem('okta-token-storage'); // ensure that the dev token is used
+    setAuthState({
+      ...authState,
```

Comment:
If we're passing a function through `setAuthState`, I think we should also pass a function here too and leverage the previous auth state, unless there's a reason why one needs to be a function and the other needs to be an object.